lib/beaker-puppet/install_utils/foss_utils.rb

```diff
@@ -438,11 +437,7 @@ def configure_puppet_on(hosts, opts = {})
         # @api private
         def install_puppet_from_rpm_on( hosts, opts )
           block_on hosts do |host|
-            if host[:type] == 'aio'
```

Comment:
@melissa if I'm understanding your changes right, we're moving to a situation where we can have `aio` runs where there are multiple different artifacts that could be installed, right? Typically those would be `pc1` for Puppet 4 artifacts, `nil` for the latest Puppet artifacts, or specific version numbers if those are wanted specifically in the new release structure, is that right?

If that's the case, we can see from [this code](https://github.com/puppetlabs/beaker/commit/688a1c8b4042723fb578f2ded19e63c8e0c32ed4) that people have been using this method to install `pc1` artifacts. The change you've proposed takes this ability away from them, as there's no way to pass it in anymore. I'm OK with de-coupling the `aio` run type from the `repo_name` parameter, but we'll have to provide some ability to still pass `pc1` to the `install_puppetlabs_release_repo` method from this one. I'm fine with an `opts[:puppet_collection]` parameter like we've used elsewhere (even though we're including things that aren't collections anymore (😞 in naming)).